experimental/CliffordAlgebras/src/CliffordAlgebras.jl

```diff
@@ -0,0 +1,584 @@
+export CliffordAlgebra,
+  CliffordAlgebraElem,
+  clifford_algebra,
+  even_coefficients,
+  odd_coefficients,
+  even_part,
+  odd_part,
+  centroid,
+  quadratic_discriminant,
+  disq
+
+################################################################################
+#
+#  Datatypes
+#
+################################################################################
+
+### Algebra ###
+mutable struct CliffordAlgebra{T,S} <: Hecke.AbstractAssociativeAlgebra{T}
+  base_ring::Ring
+  space::Hecke.QuadSpace{K,S} where {K}
+  gram::S
+  dim::Int
+  centroid::Any
+  disq::T
+  center::Any
+
+  #Return the Clifford algebra of the quadratic space 'qs' 
+  function CliffordAlgebra{T,S}(qs::Hecke.QuadSpace{K,S}) where {T,S,K}
+    gram = gram_matrix(qs)
+    return new{T,S}(base_ring(qs), qs, gram, 2^ncols(gram))
+  end
+end
+
+### Elements ###
+mutable struct CliffordAlgebraElem{T,S,P} <: Hecke.AbstractAssociativeAlgebraElem{T}
+  parent::P
```

Comment:
I think you can get rid of the third type param here and instead use
```suggestion
  parent::CliffordAlgebra{T, S}
```
This would make the code a bit simpler and thus easier to understand and maintain

Reply:
Done in this [commit](https://github.com/oscar-system/Oscar.jl/pull/4370/commits/335fc8a255bb8159870818f9ac437c04dd832981).